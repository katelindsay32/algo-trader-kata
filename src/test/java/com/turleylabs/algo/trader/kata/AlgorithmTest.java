package com.turleylabs.algo.trader.kata;

import org.approvaltests.Approvals;
import org.junit.Test;

public class AlgorithmTest {

    @Test
    public void algorithmExecutesTrades() {
        RefactorMeAlgorithm refactorAlgorithm = new RefactorMeAlgorithm();

        refactorAlgorithm.run();

        Approvals.verify(refactorAlgorithm);
    }

}
