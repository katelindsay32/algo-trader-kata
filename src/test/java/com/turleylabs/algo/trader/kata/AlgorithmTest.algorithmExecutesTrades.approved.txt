RefactorMeAlgorithm{symbol='TQQQ', movingAverage200=SimpleMovingAverage{symbol='nullvalue='73.119589225isReady='true}, movingAverage50=SimpleMovingAverage{symbol='nullvalue='96.6379794isReady='true}, movingAverage21=SimpleMovingAverage{symbol='nullvalue='100.1728559047619isReady='true}, movingAverage10=SimpleMovingAverage{symbol='nullvalue='87.0489997isReady='true}, previousMovingAverage50=96.6379794, previousMovingAverage21=100.1728559047619, previousMovingAverage10=87.0489997, previousPrice=81.089996, previous=2020-03-05, lastVix=CBOE{close='39.619999}, boughtBelow50=false, tookProfits=false, portfolio={}, trades={Trade{symbol='TQQQaveragePrice='3.206967numberOfShares='31182date='2011-01-04}Trade{symbol='TQQQaveragePrice='3.514737numberOfShares='31182date='2011-01-14}Trade{symbol='TQQQaveragePrice='3.417689numberOfShares='32067date='2011-01-25}Trade{symbol='TQQQaveragePrice='3.848069numberOfShares='32067date='2011-02-16}Trade{symbol='TQQQaveragePrice='3.496243numberOfShares='35293date='2011-04-07}Trade{symbol='TQQQaveragePrice='3.391505numberOfShares='35293date='2011-05-25}Trade{symbol='TQQQaveragePrice='3.580199numberOfShares='33433date='2011-07-11}Trade{symbol='TQQQaveragePrice='3.58477numberOfShares='33433date='2011-07-27}Trade{symbol='TQQQaveragePrice='3.457589numberOfShares='34663date='2012-01-20}Trade{symbol='TQQQaveragePrice='4.676203numberOfShares='34663date='2012-03-13}Trade{symbol='TQQQaveragePrice='4.068143numberOfShares='39843date='2012-06-18}Trade{symbol='TQQQaveragePrice='4.626744numberOfShares='39843date='2012-10-11}Trade{symbol='TQQQaveragePrice='4.388175numberOfShares='42009date='2012-11-30}Trade{symbol='TQQQaveragePrice='4.039881numberOfShares='42009date='2012-12-28}Trade{symbol='TQQQaveragePrice='4.636719numberOfShares='36602date='2013-01-03}Trade{symbol='TQQQaveragePrice='5.187006numberOfShares='36602date='2013-06-24}Trade{symbol='TQQQaveragePrice='6.068133numberOfShares='31287date='2013-07-10}Trade{symbol='TQQQaveragePrice='8.399794numberOfShares='31287date='2013-10-18}Trade{symbol='TQQQaveragePrice='8.7963numberOfShares='29876date='2013-11-13}Trade{symbol='TQQQaveragePrice='9.477095numberOfShares='29876date='2013-11-29}Trade{symbol='TQQQaveragePrice='9.615082numberOfShares='29447date='2013-12-18}Trade{symbol='TQQQaveragePrice='9.020738numberOfShares='29447date='2014-02-03}Trade{symbol='TQQQaveragePrice='10.774676numberOfShares='24654date='2014-02-19}Trade{symbol='TQQQaveragePrice='10.769689numberOfShares='24654date='2014-04-01}Trade{symbol='TQQQaveragePrice='10.265952numberOfShares='25863date='2014-05-12}Trade{symbol='TQQQaveragePrice='12.472508numberOfShares='25863date='2014-10-10}Trade{symbol='TQQQaveragePrice='15.324175numberOfShares='21050date='2014-11-04}Trade{symbol='TQQQaveragePrice='14.614169numberOfShares='21050date='2014-12-16}Trade{symbol='TQQQaveragePrice='15.857281numberOfShares='19400date='2015-02-04}Trade{symbol='TQQQaveragePrice='17.901304numberOfShares='19400date='2015-07-10}Trade{symbol='TQQQaveragePrice='20.437723numberOfShares='16992date='2015-07-22}Trade{symbol='TQQQaveragePrice='17.460548numberOfShares='16992date='2015-08-20}Trade{symbol='TQQQaveragePrice='16.407728numberOfShares='18082date='2015-10-13}Trade{symbol='TQQQaveragePrice='18.391954numberOfShares='18082date='2015-12-11}Trade{symbol='TQQQaveragePrice='15.356567numberOfShares='21657date='2016-03-03}Trade{symbol='TQQQaveragePrice='15.085462numberOfShares='21657date='2016-05-04}Trade{symbol='TQQQaveragePrice='17.387367numberOfShares='18789date='2016-06-01}Trade{symbol='TQQQaveragePrice='16.732054numberOfShares='18789date='2016-06-23}Trade{symbol='TQQQaveragePrice='17.690075numberOfShares='17772date='2016-07-13}Trade{symbol='TQQQaveragePrice='18.900904numberOfShares='17772date='2016-11-03}Trade{symbol='TQQQaveragePrice='21.231083numberOfShares='15821date='2016-11-25}Trade{symbol='TQQQaveragePrice='36.980171numberOfShares='15821date='2017-06-02}Trade{symbol='TQQQaveragePrice='36.524448numberOfShares='16018date='2017-07-18}Trade{symbol='TQQQaveragePrice='54.151321numberOfShares='16018date='2018-01-12}Trade{symbol='TQQQaveragePrice='56.483158numberOfShares='15357date='2018-03-06}Trade{symbol='TQQQaveragePrice='62.031685numberOfShares='15357date='2018-03-12}Trade{symbol='TQQQaveragePrice='55.631588numberOfShares='17123date='2018-05-14}Trade{symbol='TQQQaveragePrice='58.329342numberOfShares='17123date='2018-07-05}Trade{symbol='TQQQaveragePrice='64.686203numberOfShares='15441date='2018-07-16}Trade{symbol='TQQQaveragePrice='55.654877numberOfShares='15441date='2018-10-10}Trade{symbol='TQQQaveragePrice='43.678699numberOfShares='19674date='2019-01-17}Trade{symbol='TQQQaveragePrice='58.86871numberOfShares='19674date='2019-05-15}Trade{symbol='TQQQaveragePrice='61.917606numberOfShares='18705date='2019-06-19}Trade{symbol='TQQQaveragePrice='55.220032numberOfShares='18705date='2019-08-05}Trade{symbol='TQQQaveragePrice='61.177876numberOfShares='16884date='2019-09-04}Trade{symbol='TQQQaveragePrice='58.918694numberOfShares='16884date='2019-10-03}Trade{symbol='TQQQaveragePrice='65.856178numberOfShares='15105date='2019-10-17}Trade{symbol='TQQQaveragePrice='94.330002numberOfShares='15105date='2020-01-09}Trade{symbol='TQQQaveragePrice='98.480003numberOfShares='14468date='2020-01-30}Trade{symbol='TQQQaveragePrice='104.629997numberOfShares='14468date='2020-02-04}}}